server/controllers/users.js

```diff
@@ -52,30 +52,31 @@ export const exists = async (req, res) => {
 /**
  * Login or create a new user
  */
-export const signin = (req, res, next) => {
-  const { user, redirect, websiteUrl } = req.body;
-  let loginLink;
-  let clientIP;
-  return models.User.findOne({ where: { email: user.email.toLowerCase() } })
-    .then(u => u || models.User.createUserWithCollective(user))
-    .then(u => {
-      loginLink = u.generateLoginLink(redirect || '/', websiteUrl);
-      clientIP = req.ip;
-      if (config.env === 'development') {
-        logger.info(`Login Link: ${loginLink}`);
-      }
-      return emailLib.send('user.new.token', u.email, { loginLink, clientIP }, { sendEvenIfNotProduction: true });
-    })
-    .then(() => {
-      const response = { success: true };
-      // For e2e testing, we enable testuser+(admin|member)@opencollective.com to automatically receive the login link
-      if (config.env !== 'production' && user.email.match(/.*test.*@opencollective.com$/)) {
-        response.redirect = loginLink;
-      }
-      return response;
-    })
-    .then(response => res.send(response))
-    .catch(next);
+export const signin = async (req, res) => {
+  const { user, redirect, websiteUrl, createProfile = true } = req.body;
```

Comment:
In which case do we have createProfile set to TRUE, set to FALSE?

Reply:
We should add a code comment and delete this behavior in the future as it's not necessary.

Reply:
I've added a comment. This function needs to be refactored after we fully remove the `signin.js` on front-end (remove the old signin flow).